test/RachisTests/DatabaseCluster/ReplicationTests.cs

```diff
@@ -674,20 +674,49 @@ public async Task AddGlobalChangeVectorToNewDocument(bool useSsl)
                     await session.SaveChangesAsync();
                 }
 
+                Assert.True(await WaitForDocumentInClusterAsync<User>(
+                    databaseResult.Topology,
+                    databaseName,
+                    "users/1",
+                    u => u.Name.Equals("Karmel"),
+                    TimeSpan.FromSeconds(60),
+                    certificate: clientCertificate));
+
                 // we need to wait for database change vector to be updated
                 // which means that we need to wait for replication to do a full mesh propagation
-                Assert.True(await WaitForValueOnGroupAsync(topology, serverStore =>
+                try
                 {
-                    var database = serverStore.DatabasesLandlord.TryGetOrCreateResourceStore(databaseName).Result;
+                    await WaitForValueOnGroupAsync(topology, serverStore =>
+                    {
+                        var database = serverStore.DatabasesLandlord.TryGetOrCreateResourceStore(databaseName).Result;
 
-                    using (database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
-                    using (context.OpenReadTransaction())
+                        using (database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+                        using (context.OpenReadTransaction())
+                        {
+                            var cv = DocumentsStorage.GetDatabaseChangeVector(context);
+
+                            return cv != null && cv.Contains("A:1-") && cv.Contains("B:1-") && cv.Contains("C:1-");
+                        }
+                    }, expected: true, timeout: 60000);
+                }
+                catch (Exception e)
+                {
+                    var error = e.Message;
+                    foreach (var node in topology.AllNodes)
                     {
-                        var cv = DocumentsStorage.GetDatabaseChangeVector(context);
+                        var serverStore = Servers.Single(s => s.ServerStore.NodeTag == node).ServerStore;
+                        var database = serverStore.DatabasesLandlord.TryGetOrCreateResourceStore(databaseName).Result;
+                        using (database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+                        using (context.OpenReadTransaction())
+                        {
+                            var cv = DocumentsStorage.GetDatabaseChangeVector(context);
 
-                        return cv != null && cv.Contains("A:1-") && cv.Contains("B:1-") && cv.Contains("C:1-");
+                            error += $" {node}: cv.";
```

Comment:
Should be `$"{node}:{cv}"` also, let's get the outgoing failures. 
Something similar to the endpoint `"/databases/*/replication/debug/outgoing-failures"`